src/main/java/de/tum/in/www1/artemis/web/rest/TextSubmissionResource.java

```diff
@@ -203,12 +202,13 @@ else if (!authorizationCheckService.isAtLeastInstructorForExercise(exercise)) {
      * GET /text-submission-without-assessment : get one textSubmission without assessment.
      *
      * @param exerciseId exerciseID  for which a submission should be returned
+     * @param lockSubmission optional value to define if the submission should be locked and has the value of false if not set manually
      * @return the ResponseEntity with status 200 (OK) and the list of textSubmissions in body
      */
     @GetMapping(value = "/exercises/{exerciseId}/text-submission-without-assessment")
     @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
-    @Transactional(readOnly = true)
```

Comment:
I can't remove the transaction here yet, as it throws an exception inside `calculateSmallerClusterPercentageBatch` method in `TextAssessmentQueueService` We need to fetch dependencies in a very nested nature. Maybe @jpbernius you can help me with this? Otherwise let's leave `Transactional` here